### PR TITLE
build: cleanup build

### PR DESCRIPTION
**Pull request by alonbl**
cleanup maven trivial statements.

use java.level instead of java.version to actually use java8.

resolve findbugs issues

resolve java issues

### CODE REVIEW

**Comment by alonbl** [comment]
@erikhakansson: I've checked the minimum requirements for adding the apache-sshd, it seems to work without additional statements you added in your patch, can you please checkout? Just add the apache-sshd dependency in your patch and use this as a base.

File: src/main/java/jenkins/plugins/gerrit/AbstractGerritSCMSource.java [hunk 1]
```diff
@@ -183,16 +182,16 @@ private boolean isOpenChange(String refName, HashSet<Integer> openChanges) {
...
   protected List<Action> retrieveActions(
-      @CheckForNull SCMSourceEvent event, @NonNull TaskListener listener)
+      @CheckForNull SCMSourceEvent event, @Nonnull TaskListener listener)
       throws IOException, InterruptedException {
     return doRetrieve(
         new Retriever<List<Action>>() {
           @Override
           public List<Action> run(
-              GitClient client, String remoteName, Changes.QueryRequest queryRequest)
+              GitClient client, GitSCMSourceContext context, String remoteName, Changes.QueryRequest queryRequest)
```
Comment (lucamilanesio) [inline src/main/java/jenkins/plugins/gerrit/AbstractGerritSCMSource.java]: This seems a bit different from the other cleanups. What was this causing in terms of user-facing problems?

**Comment by alonbl** [comment]
The CI windows slaves are offline, failure is because of them.

**Comment by lucamilanesio** [comment]
@alonbl to be rebased and then it should be good to go

**Comment by alonbl** [comment]
I asked to merge this before the format so I do not have to deal with rebase.

**Comment by alonbl** [comment]
Rebased

**Comment by alonbl** [comment]
@lucamilanesio : any comment regarding this? these kind of cleanup tend to be hard to maintain over time if not merged, and we should have zero warnings.

File: pom.xml [hunk 2]
```diff
@@ -135,34 +121,30 @@
...
+    </developer>
+    <developer>
+      <id>alonbl</id>
+      <name>Alon Bar-Lev</name>
+      <email>alon.barlev@gmail.com</email>
+    </developer>
+  </developers>
 
   <dependencies>
-    <dependency>
```
Comment (lucamilanesio) [inline pom.xml]: Is this related to the java.level change?

**Comment by alonbl** [comment]
> Let me try to import this into GerritHub and we can continue the review there 😄

can we please finish this one and the #41 here before moving around?

**Comment by lucamilanesio** [comment]
Review imported to GerritHub (https://review.gerrithub.io/c/jenkinsci/gerrit-code-review-plugin/+/409176)

**Pull request closed without merging**